client/src/main/java/io/pravega/client/stream/impl/LargeEventWriter.java

```diff
@@ -0,0 +1,252 @@
+                write(segment, payloads, client, tokenProvider);
+                sent = true;
+            } catch (ConnectionFailedException e) {
+                log.info("Connection failure while sending large event: {}. Retrying", e.getMessage());
+                cause = e;
+            } catch (TokenExpiredException e) {
+                tokenProvider.signalTokenExpired();
+                log.info("Authentication token expired while writing large event to segment {}. Retrying", segment);
+            }
+            attempts--;
+        }
+        if (!sent) {
+            throw new RetriesExhaustedException("Failed to write large event to segment " + segment, cause);
+        }
+    }
+
+    private List<ByteBuf> createBufs(List<ByteBuffer> events) {
+        ByteBuffer[] toWrite = new ByteBuffer[2 * events.size()];
+        for (int i = 0; i < events.size(); i++) {
+            ByteBuffer event = events.get(i);
+            byte[] header = new byte[WireCommands.TYPE_PLUS_LENGTH_SIZE];
+            ByteBuffer wrapped = ByteBuffer.wrap(header);
+            wrapped.putInt(WireCommandType.EVENT.getCode());
+            wrapped.putInt(event.remaining());
+            wrapped.flip();
+            toWrite[2 * i] = wrapped;
+            toWrite[2 * i + 1] = event;
+        }
+        ByteBuf master = Unpooled.wrappedBuffer(toWrite);
+        ArrayList<ByteBuf> result = new ArrayList<>();
+        while (master.isReadable()) {
+            int toRead = Math.min(master.readableBytes(), WRITE_SIZE);
+            result.add(master.readSlice(toRead));
+        }
+        return result;
+    }
+
+    private void write(Segment parentSegment, List<ByteBuf> payloads, RawClient client,
+            DelegationTokenProvider tokenProvider) throws TokenExpiredException, NoSuchSegmentException,
+            AuthenticationException, SegmentSealedException, ConnectionFailedException {
+
+        long requestId = client.getFlow().getNextSequenceNumber();
+        log.debug("Writing large event to segment {} with writer id {}", parentSegment, writerId);
+
+        String token = getThrowingException(tokenProvider.retrieveToken());
+
+        CreateTransientSegment createSegment = new CreateTransientSegment(requestId,
+                writerId,
+                parentSegment.getScopedName(),
+                token);
+
+        SegmentCreated created = transformSegmentCreated(getThrowingException(client.sendRequest(requestId, createSegment)),
+                                                         parentSegment.getScopedName());
+        requestId = client.getFlow().getNextSequenceNumber();
+        SetupAppend setup = new SetupAppend(requestId, writerId, created.getSegment(), token);
+
+        AppendSetup appendSetup = transformAppendSetup(getThrowingException(client.sendRequest(requestId, setup)),
+                                                       created.getSegment());
+
+        if (appendSetup.getLastEventNumber() != 0) {
```

Comment:
I believe the `AppendProcessor` will using `Long.MIN_VALUE` by [default](https://github.com/pravega/pravega/blob/master/segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java#L185). Is that a valid value in this case?

Reply:
Fixed.